### PR TITLE
[FIX] *: SyntaxWarning: invalid escape sequence

### PR DESCRIPTION
**Pull request by Xavier-Do**
Since python 3.12 invalid escape sequences are (even more) deprecated and this will be removed in future versions. This commit fixes all the missing escape **automatically** using `ruff check --select W605 --fix`. Some will be escaped, other may be replaced by r string. No manual adaptation will be done in this pr

part of #162438

### CODE REVIEW

File: addons/account/models/account_account_tag.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def _get_tax_tags(self, tag_name, country_id):
         """ Returns all the tax tags corresponding to the tag name given in parameter
         in the specified country.
         """
-        escaped_tag_name = tag_name.replace('\\', '\\\\').replace('%', '\%').replace('_', '\_')
+        escaped_tag_name = tag_name.replace('\\', '\\\\').replace('%', r'\%').replace('_', r'\_')
```
Comment (rdeodoo) [inline addons/account/models/account_account_tag.py:23]:
```suggestion
        escaped_tag_name = tag_name.replace('\\', '\\\\').replace('%', '\\%').replace('_', '\\_')
```
Shouldn't this be done like this? Those are not really regex strings, they are just wrongly escaped no?
It's doing the same with double `\` but prevent the linter error
```py
>>> "a_b".replace('_', '\_')
'a\\_b'
>>> "a_b".replace('_', '\\_')
'a\\_b'
```

File: addons/mail/wizard/mail_compose_message.py [hunk 2]
```diff
@@ -559,7 +559,7 @@ def _get_optout_emails(self, mail_values_dict):
         return []
 
     def _onchange_template_id(self, template_id, composition_mode, model, res_id):
-        """ - mass_mailing: we cannot render, so return the template values
+        r""" - mass_mailing: we cannot render, so return the template values
```
Comment (rdeodoo) [inline addons/mail/wizard/mail_compose_message.py:562]: Isn't it weird to mark docstring as regex string just for `/!\` to not raise the linter error? Wouldn't a `# noqa:` notation be more logical?

File: addons/web/controllers/main.py [hunk 3]
```diff
@@ -367,7 +367,7 @@ def xml2json_from_elementtree(el, preserve_whitespaces=False):
     return res
 
 class HomeStaticTemplateHelpers(object):
-    """
+    r"""
```
Comment (rdeodoo) [inline addons/web/controllers/main.py:370]: Here too (and other places). It's a docstring, not a regex? Maybe worth changing `/!\` into `!!!` to indirectly simply bypass it in such a case?

File: odoo/tools/sql.py [hunk 4]
```diff
@@ -249,7 +249,7 @@ def drop_view_if_exists(cr, viewname):
         cr.execute("DROP MATERIALIZED VIEW {} CASCADE".format(viewname))
 
 def escape_psql(to_escape):
-    return to_escape.replace('\\', r'\\').replace('%', '\%').replace('_', '\_')
+    return to_escape.replace('\\', r'\\').replace('%', r'\%').replace('_', r'\_')
```
Comment (rdeodoo) [inline odoo/tools/sql.py:252]:
```suggestion
    return to_escape.replace('\\', r'\\').replace('%', '\\%').replace('_', '\\_')
```
To confirm but same as above

File: odoo/addons/base/tests/test_mail.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def test_evil_malicious_code(self):
             ("<DIV STYLE=\"background-image: url(&#1;javascript:alert('XSS'))\">"),  # div background + extra characters
             ("<IMG SRC='vbscript:msgbox(\"XSS\")'>"),  # VBscrip in an image
             ("<BODY ONLOAD=alert('XSS')>"),  # event handler
-            ("<BR SIZE=\"&{alert('XSS')}\>"),  # & javascript includes
+            ("<BR SIZE=\"&{alert('XSS')}\\>"),  # & javascript includes
```
Comment (Julien00859) [inline odoo/addons/base/tests/test_mail.py:67]: you missed one
```suggestion
            ("<BR SIZE=\\"&{alert('XSS')}\\>"),  # & javascript includes
```
```suggestion
            (r"<BR SIZE=\"&{alert('XSS')}\>"),  # & javascript includes
```

File: odoo/addons/base/tests/test_qweb.py [hunk 6]
```diff
@@ -1342,8 +1342,8 @@ def test_no_empty_lines(self):
             'arch_db': self.arch
         })
         rendered = self.env['ir.qweb']._render(t.id)
-        self.assertFalse(re.compile('^\s+\n').match(rendered))
-        self.assertFalse(re.compile('\n\s+\n').match(rendered))
+        self.assertFalse(re.compile('^\\s+\n').match(rendered))
+        self.assertFalse(re.compile('\n\\s+\n').match(rendered))
```
Comment (Julien00859) [inline odoo/addons/base/tests/test_qweb.py:1346]: `\n` is recognized by the regexp engine as well
```suggestion
        self.assertFalse(re.compile(r'^\s+\n').match(rendered))
        self.assertFalse(re.compile(r'\n\s+\n').match(rendered))
```

File: odoo/addons/base/tests/test_qweb.py [hunk 7]
```diff
@@ -1352,8 +1352,8 @@ def test_keep_empty_lines(self):
             'arch_db': self.arch
         })
         rendered = self.env['ir.qweb']._render(t.id, {'__keep_empty_lines': True})
-        self.assertTrue(re.compile('^\s+\n').match(rendered))
-        self.assertTrue(re.compile('\n\s+\n').match(rendered))
+        self.assertTrue(re.compile('^\\s+\n').match(rendered))
+        self.assertTrue(re.compile('\n\\s+\n').match(rendered))
```
Comment (Julien00859) [inline odoo/addons/base/tests/test_qweb.py:1356]: ditto
```suggestion
        self.assertTrue(re.compile(r'^\s+\n').match(rendered))
        self.assertTrue(re.compile(r'\n\s+\n').match(rendered))
```

**Review by Julien00859 — CHANGES_REQUESTED** [review]

File: addons/account/models/account_move.py [hunk 8]
```diff
@@ -5091,7 +5091,7 @@ def _mail_track(self, tracked_fields, initial):
     # -------------------------------------------------------------------------
 
     def _prepare_reconciliation_partials(self):
-        ''' Prepare the partials on the current journal items to perform the reconciliation.
+        r''' Prepare the partials on the current journal items to perform the reconciliation.
         /!\ The order of records in self is important because the journal items will be reconciled using this order.
```
Comment (oomsveta) [inline addons/account/models/account_move.py:5095]: It seems a bit strange to me to have to use a raw string as docstring just to be able to write /!\\. How about replacing the warning signs with the correct codepoint instead? https://unicode-explorer.com/c/26A0

File: addons/mail/wizard/mail_compose_message.py (hunk 2 above)
Comment (Xavier-Do) [reply addons/mail/wizard/mail_compose_message.py:562]: It not a linting issue but a parsing issue, in python 3.14 this code won't even run.

File: odoo/tools/sql.py (hunk 4 above)
Comment (Xavier-Do) [reply odoo/tools/sql.py:252]: I'm sorry but I wont handfix that in all versions, if the script behaviour is correct I'll do the automated way in all versions. 
I can wait a little if you want to fix your cases by hand before I start merging this pr.

File: addons/base_setup/tests/test_res_config_doc_links.py [hunk 9]
```diff
@@ -17,7 +17,7 @@ def setUp(self):
         Set-up the test environment
         """
         super(TestResConfigDocLinks, self).setUp()
-        self.re = re.compile("<a href=\"(\S+/documentation/\S+)\"")
+        self.re = re.compile("<a href=\"(\\S+/documentation/\\S+)\"")
```
[the remark this answers is not in the document]
Comment (Xavier-Do) [reply addons/base_setup/tests/test_res_config_doc_links.py:20]: I don't want to fix them by hand for style reasons if the automatic fix is correct.

File: addons/mail/wizard/mail_compose_message.py (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/mail/wizard/mail_compose_message.py:562]: Why don't you remove /!\ then ?

**Comment by Xavier-Do** [comment]
General answer about nitpicking:

This change is needed, mostly in following versions, because it won't work in python 3.14.

This change is automated, there is too much to do to fix it by hand without fearing to make a mistake. 
This fix will most likely conflict in all versions, and individual pr may be created applying the ruff autopatch without forwardport.

I won't apply individual suggestions. If the fix is correct, but you don't like it, you are welcome to fix it yourself in another pr.

If you want to suggest another way to **autopatch**, say it, but since ruff is the tool used for linting it makes sence to follow it's guidelines.

**Comment by Xavier-Do** [comment]
@fw-bot disable (will conflict, forwardporting manually)

**Comment by d-fence** [comment]
robodoo r+

**Pull request closed without merging**